mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/ElasticSearchIntegrationTest.java

```diff
@@ -218,4 +218,28 @@ public void reasonableLongTermShouldNotBeIgnored() throws Exception {
         assertThat(messageManager.search(new SearchQuery(SearchQuery.bodyContains(reasonableLongTerm)), session))
             .containsExactly(composedMessageId.getUid());
     }
+
+    @Test
+    public void dynamicMappingShouldFailMessageIndexationWithDifferentTypesOnSameCustomField() throws Exception {
```

Comment:
This is more `headerSearchShouldIncludeMessageWhenDifferentTypesOnADinamicallyIndexedField`